app/markets/ActivityTable.tsx

```diff
@@ -33,33 +43,54 @@ const TAG_COLOR_SCHEMES: { 0: TagColorSchemeProp; 1: TagColorSchemeProp } = {
 
 const ITEMS_PER_PAGE = 10;
 
-export const ActivityTable = ({ id }: { id: string }) => {
-  const [page, setPage] = useState(1);
+// equal fields on Trades and Trasanctions
+//    id, transactionHash, creationTimestamp
 
-  const { data: trades, isLoading } = useQuery({
-    queryKey: ['getMarketTrades', id, page - 1],
-    queryFn: async () =>
-      getMarketTrades({
-        first: ITEMS_PER_PAGE,
-        skip: (page - 1) * ITEMS_PER_PAGE,
-        fpmm: id,
-        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
-        orderDirection: OrderDirection.Desc,
-      }),
+// collading, same name, diff
+// fpmm
+//    trade -> outcomes: ["yes", "no"]
+//    tx -> collateralToken:  "0xe91d153e0b41518a2ce8dd3d7944fa863463a97d", __typename: "FixedProductMarketMaker"
+// fields only on trade
+//    creator, fpmm, id, outcomeIndex, outcomeTokensTraded
+
+// fields only on transaction
+//    additionalSharesCost, collateralTokenAmount, fpmmType, sharesOrPoolTokenAmount, transactionType, user, __typename
+
+function mergeMarketTradesTransactionsArrays(
+  trades: FpmmTrade[],
+  transactions: FpmmTransaction[]
+): (FpmmTrade & Partial<FpmmTransaction>)[] {
+  const tradesAndTransactions = new Map<string, FpmmTrade & Partial<FpmmTransaction>>();
+  // setup all trades
+  trades.forEach(trade => {
+    tradesAndTransactions.set(trade.id, { ...trade });
   });
 
-  const { data: tradesNextPage } = useQuery({
-    queryKey: ['getMarketTrades', id, page],
-    queryFn: async () =>
-      getMarketTrades({
-        first: ITEMS_PER_PAGE,
-        skip: page * ITEMS_PER_PAGE,
-        fpmm: id,
-        orderBy: FpmmTrade_OrderBy.CreationTimestamp,
-        orderDirection: OrderDirection.Desc,
-      }),
+  // Process all transactions
+  transactions.forEach(transaction => {
+    // if trade has a transaction with same id
+    if (tradesAndTransactions.has(transaction.id)) {
+      const trade = tradesAndTransactions.get(transaction.id)!;
+      tradesAndTransactions.set(transaction.id, {
+        ...trade,
+        ...transaction,
+        fpmm: trade.fpmm,
+        collateralToken: transaction.fpmm?.collateralToken,
+      });
+    } else if (transaction.fpmmType.toLowerCase() === 'liquidity') {
```

Comment:
You can get 'liquidity' from `transaction.fpmmType` and no need for lower casing.

Reply:
Yes. Thanks.